crypto/x509/x509type.c

```diff
@@ -41,6 +41,9 @@ int X509_certificate_type(const X509 *x, const EVP_PKEY *pkey)
     case EVP_PKEY_EC:
         ret = EVP_PK_EC | EVP_PKT_SIGN | EVP_PKT_EXCH;
         break;
+    case NID_ED25519:
+        ret = EVP_PKT_SIGN;
```

Comment:
For completeness, I guess we should, although I can't imagine there are ever going to be too many X25519 certificates???